### PR TITLE
Update docs to recommend blobless cloning strategy

### LINKED ISSUE #4329: Dramatically improve cloning speed for contributors
## Description

I did some analysis on the size of the repository, originally intended to address concerns about the size of our frontend snapshots, that led to some insights. First, a breakdown of the current repo sizing:

- Downloaded repo size: 1.4GB
- Side of repo `.git` directory: 1.3GB

It also took over 2 minutes to download on my 120 Mbps wireless connection.

Clearly, the history and metadata of the repository are the main contributors to the download size. Additionally, there aren't really any large blobs in particular that we benefit from removing. We simply have a lot of history with a lot of files.

I think for most contributors we should recommend doing a "Partial blobless clone" of the repository using the flag `--filter=blob:none` like so:

```shell
git clone --filter=blob:none https://github.com/wordpress/openverse.git
# or gh repo clone wordpress/openverse -- --filter=blob:none 
```

This results in the following sizing:

- Downloaded repo size: 183MB
- Side of repo `.git` directory: 79M

with a 15 second download time on my 120 Mbps wireless connection.

You can learn more about blobless clones here: https://gist.github.com/leereilly/1f4ea46a01618b6e34ead76f75d0784b#blobless-clones

It basically means that all of the metadata of past commits are present, but not the actual files (blobs). Those will be downloaded on-demand when running `git blame` or `git checkout` of a previous commit.

I think we can recommend this strategy to most users in our documentation, and significantly improve their experience.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #4329 by @zackkrida 

## Description

Update our quickstart guides to recommend blobless cloning, saving contributors much data and time. Success!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: documentation/api/guides/quickstart.md [hunk 1]
```diff
@@ -17,8 +17,12 @@ you need to run this.
 2. Clone the repository to your computer. Then switch to the cloned directory.
    If you're planning to contribute, fork the repo and clone your fork instead.
 
+   ```{note}
+   We recommend cloning with the `--filter=blob:none` flag as it dramatically reduces the filesize and download time by creating a "blobless clone". You can learn more about these [here](https://gist.github.com/leereilly/1f4ea46a01618b6e34ead76f75d0784b).
```
Comment (krysal) [inline documentation/api/guides/quickstart.md:21]: Can this and the others notes in quickstart documents be wrapped?
```suggestion
   We recommend cloning with the `--filter=blob:none` flag as it dramatically
   reduces the filesize and download time by creating a "blobless clone".
   You can learn more about these [here](https://gist.github.com/leereilly/1f4ea46a01618b6e34ead76f75d0784b).
```

File: documentation/catalog/guides/quickstart.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ project, see [DAGs.md](../reference/DAGs.md).
 See each provider API script's notes in their respective [handbook][ov-handbook]
 entry.
 
-[ov-handbook]: https://make.wordpress.org/openverse/handbook/
+[ov-handbook]: https://make.WordPress.org/openverse/handbook/
```
Comment (krysal) [inline documentation/catalog/guides/quickstart.md:24]: This change is interesting, was it done by the linter?

File: documentation/catalog/guides/quickstart.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ project, see [DAGs.md](../reference/DAGs.md).
 See each provider API script's notes in their respective [handbook][ov-handbook]
```
Comment (AetherUnbound) [inline documentation/catalog/guides/quickstart.md:21]: It looks like the changes to this file were made automatically, should they be reverted? 😅

**Review by AetherUnbound — APPROVED** [review]
LGTM barring some unnecessary changes to the catalog quickstart

**Review by krysal — APPROVED** [review]
LGTM! Great addition to the docs 💯

File: documentation/api/guides/quickstart.md (hunk 1 above)
Comment (zackkrida) [reply documentation/api/guides/quickstart.md:21]: @krysal sure, I'll do that. I was concerned the wrapping would be preserved on the web but apparently that isn't the case for these "admonition" blocks!

**Review by dhruvkb — APPROVED** [review]
I guess we should not have copied the clone step 5 times!

**Pull request merged**